docs/requiredtools/typescript.md

```diff
@@ -7,12 +7,188 @@ parent: Required tools
+
+```PowerShell
+Connect-MgGraph -Scopes "User.Read", "Application.ReadWrite.All"
+$app = New-MgApplication -DisplayName "Kiota - NodeJS" `
+  -IsFallbackPublicClient `
+  -SignInAudience AzureADMyOrg `
+  -RequiredResourceAccess @{ "ResourceAppId" = "00000003-0000-0000-c000-000000000000"; "ResourceAccess" = @( @{ Id = "e1fe6dd8-ba31-4d61-89e7-88639da4683d"; Type = "Scope"}) }
+
+Write-Host "Your Tenant Id is : $((Get-MgOrganization).Id)"
+Write-Host "Your Client Id is : $($app.AppId)"
+Disconnect-MgGraph
+```
+
+Record the value of the ClientId property of the $app object as it will be needed in a later step.
+
+## Creating the client application
+
+Before adding our actual code, create an `auth.ts` file at the root of the `src` folder that will handle our authentication.
+
+```typescript
+import { DeviceCodeCredential } from "@azure/identity";
+import { BaseBearerTokenAuthenticationProvider } from "@microsoft/kiota-abstractions"
+
+export class Auth extends BaseBearerTokenAuthenticationProvider {
+   getAuthorizationToken = async (): Promise<string> => {
+      const tenantId = "REPLACE_BY_TENANT_ID"; 
+      const clientId = "REPLACE_BY_CLIENT_ID"; 
+      const scopes = "https://graph.microsoft.com/.default";
+
+      const credentials = new DeviceCodeCredential({ tenantId: tenantId, clientId: clientId});
+      const token = (await credentials.getToken(scopes)).token;
+
+      return Promise.resolve(token);
+    }
+}
+```
+
+Then, change the `noUnusedLocals` value to `false` in the tsconfig.json at the root of your project folder.
+
+```json
+{
+  "compilerOptions": {
+    "target": "ES2020",
+    "module": "ES2020",
+    "moduleResolution": "node",
+    "lib": ["dom", "es2020"],
+    "allowSyntheticDefaultImports": true,
+    "experimentalDecorators": true,
+    "noUnusedLocals": true,
+    "removeComments": true,
+    "strict": true,
+    "typeRoots": ["node_modules/@types"]
+  },
+  "include": ["src/**/*.ts"],
+  "exclude": ["node_modules"],
+}
+```
+
+The final step is to update the main.ts file that was generated as part of the console application to include the code below.
+
+```typescript
+import { App } from './app/app';
+import { Logger } from './app/common/logger';
+import { GraphServiceClient } from "./kiota/graphServiceClient";
+import { FetchRequestAdapter } from "@microsoft/kiota-http-fetchlibrary";
+import { Auth } from "./auth"
+import { User } from './kiota/models/microsoft/graph/user';
+
+Logger.logTask('SYSTEM', 'STARTING');
+
+App.run();
+
+const core = new FetchRequestAdapter(new Auth());
+const client = new GraphServiceClient(core);
+
+async function getMe(): Promise<User | undefined> {
+    const userRequestBuilder = client.me.get();
+    return await userRequestBuilder;
+}
+
+getMe().then((user: User | undefined) => { 
```

Comment:
you might actually need to return the promise to make sure the app doesn't close before the promise returns

Reply:
Not sure what you mean? Returning the .get() instead of the builder?

Reply:
Can't do that in a non-function. In this case... As we're chaining the getMe with .then... Wouldn't this force the console to wait for it?

Reply:
no changes required if you don't run into cases where the app returns before completing the promise.
usually I make my changes in app.run method instead of the global bootstrap file, hence why I need to return I believe.